CustomItems/Items/DeflectorShield.cs

```diff
@@ -0,0 +1,149 @@
+        /// </summary>
+        [Description("How long the deflector shield can be wore, before automaticly player takes it off. (set to 0 for no limit)")]
+        public float Duration { get; set; } = 15f;
+
+        /// <summary>
+        /// Gets or sets By what will the Damage be multiplied.
+        /// </summary>
+        [Description("By what will the Damage be multiplied")]
+        public float Multiplier { get; set; } = 1;
+
+        /// <inheritdoc/>
+        protected override void SubscribeEvents()
+        {
+            Exiled.Events.Handlers.Player.MedicalItemDequipped += OnMedicalItemDeEquipped;
+            Exiled.Events.Handlers.Player.Destroying += OnDestroying;
+            Exiled.Events.Handlers.Player.Died += OnDied;
+            Exiled.Events.Handlers.Player.Hurting += OnHurt;
+
+            base.SubscribeEvents();
+        }
+
+        /// <inheritdoc/>
+        protected override void UnsubscribeEvents()
+        {
+            Exiled.Events.Handlers.Player.MedicalItemDequipped -= OnMedicalItemDeEquipped;
+            Exiled.Events.Handlers.Player.Destroying -= OnDestroying;
+            Exiled.Events.Handlers.Player.Died -= OnDied;
+            Exiled.Events.Handlers.Player.Hurting -= OnHurt;
+
+            base.UnsubscribeEvents();
+        }
+
+
+        /// <inheritdoc/>
+        protected override void OnDropping(DroppingItemEventArgs ev)
+        {
+            if (deflectorPlayers.Contains(ev.Player) && Check(ev.Item))
+            {
+                ev.IsAllowed = false;
+
+                deflectorPlayers.Remove(ev.Player);
+            }
+            else
+            {
+                base.OnDropping(ev);
+            }
+        }
+
+        /// <inheritdoc/>
+        protected override void OnWaitingForPlayers()
+        {
+            deflectorPlayers.Clear();
+
+            base.OnWaitingForPlayers();
+        }
+
+        private void OnDied(DiedEventArgs ev)
+        {
+            if (deflectorPlayers.Contains(ev.Target))
+                deflectorPlayers.Remove(ev.Target);
+        }
+
+        private void OnDestroying(DestroyingEventArgs ev)
+        {
+            if (deflectorPlayers.Contains(ev.Player))
+                deflectorPlayers.Remove(ev.Player);
+        }
+
+        private void OnMedicalItemDeEquipped(DequippedMedicalItemEventArgs ev)
+        {
+            if (!Check(ev.Player.CurrentItem))
+                return;
+
+            if (!deflectorPlayers.Contains(ev.Player))
+                deflectorPlayers.Add(ev.Player);
+
+            ev.Player.ReferenceHub.playerEffectsController.DisableEffect<Scp268>();
+
+            if (Duration > 0)
+            {
+                Timing.CallDelayed(Duration, () =>
+                {
+                    deflectorPlayers.Remove(ev.Player);
+                });
+            }
+        }
+
+        private void OnHurt(HurtingEventArgs ev)
```

Comment:
Make sure ev.Target != ev.Attacker, otherwise you will cause a stack overflow if someone damages themselves with a grenade.